src/main/java/dk/bec/polonez/reservationsystem/service/UserService.java

```diff
@@ -32,4 +50,137 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
                         new SimpleGrantedAuthority(user.getRole().getName())));
 
     }
+
+    public List<ProfileResponse> getAllUsers() throws ResponseStatusException{
+        if(!authService.isAdminLoggedIn())
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+
+        List<ProfileResponse> profiles = new ArrayList<ProfileResponse>();
+        userRepository.findAll().forEach( (n) -> profiles.add(userToResponse(n)));
+
+        return profiles;
+    }
+
+    public SignupResponse createUser(SignupRequest request) throws ResponseStatusException{
+        if (userRepository.existsByUsername(request.getUsername())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Error: Username is already taken!");
+        }
+
+        if (userRepository.existsByEmail(request.getEmail())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Error: Email is already in use!");
+        }
+
+        String strRole = request.getRole().toUpperCase();
+        Role role = roleRepository.findByName(strRole)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "Role not found"));
+
+        User user = User.builder()
+                .username(request.getUsername())
+                .password(encoder.encode(request.getPassword()))
+                .email(request.getEmail())
+                .role(role)
+                .build();
+
+        User savedUser = userRepository.save(user);
+        return SignupResponse.builder()
+                .username(savedUser.getUsername())
+                .email(savedUser.getEmail())
+                .build();
+    }
+
+    public UpdateResponse updateUser(UpdateRequest updatedProfile) throws ResponseStatusException{
+        if(!authService.isAdminLoggedIn())
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+
+
+        Optional<User> optionalUser = userRepository.findById(updatedProfile.getId());
+        User user;
+        if(optionalUser.isPresent())
```

Comment:
**advice** Instead of using ```if``` you can use a power of Java ```Optional``` type:

```
        User user = userRepository.findById(updatedProfile.getId())
            .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Error: User not found"));
```